src/client/datascience/interactive-ipynb/randomBytes.ts

```diff
@@ -24,60 +28,49 @@ export class SystemPseudoRandomNumberGenerator implements ISystemPseudoRandomNum
         }
     }
 
-    // Calls into BCryptGenRandom in bcrypt.dll using node-ffi
-    // See https://github.com/node-ffi/node-ffi/wiki/Node-FFI-Tutorial for details on usage
-    private randomBytesForWindows(numBytes: number) {
-        try {
-            // tslint:disable: no-require-imports
-            // Lazy-load modules required for calling BCryptGenRandom
-            const ffi = require('ffi-napi') as typeof import('ffi-napi');
-            const ref = require('ref-napi');
-
-            const BCRYPT_ALG_HANDLE = 'void*';
-            const ULONG = 'uint';
-            const PUCHAR = 'pointer';
-            const NTSTATUS = ref.types.uint32;
-
-            traceInfo('Initializing FFI bindings for BCryptGenRandom...');
-            const bcryptlib = ffi.Library('BCrypt', {
-                // Name of DLL function: [ return type, [ arg1 type, arg2 type, ... ] ]
-                // https://docs.microsoft.com/en-us/windows/win32/api/bcrypt/nf-bcrypt-bcryptgenrandom
-                BCryptGenRandom: [NTSTATUS, [BCRYPT_ALG_HANDLE, PUCHAR, ULONG, ULONG]]
+    // Run a small bundled executable which directly calls BCryptGenRandom and
+    // outputs 1024 random bytes to stdout as a hex string.
+    private async randomBytesForWindows(_numBytes: number): Promise<string> {
+        // Ensure the exe is present. If it's not we can't generate bytes for Windows
+        const executable = path.resolve(EXTENSION_ROOT_DIR, 'out', 'BCryptGenRandom', 'BCryptGenRandom.exe');
```

Comment:
Is there any way to check signing? Or does this happen automatically on windows?

Reply:
Are you asking if we can verify that the executable is signed before running it?

Reply:
Do we need to verify that, we're shipping the binary our selves, hence we trust what we are shipping.
If the files have been tampered with, then end user has bigger problems (machine has already been compromised)